### PR TITLE
[IndexFilters] Loading spinner moved to be a suffix within the search box.

### PR DESCRIPTION
**Pull request by jesstelford**
Related to https://github.com/Shopify/polaris-internal/issues/999

**Before**
image

**This PR**
index-filters-loading-demo

|  |  |
|---|---|
| Filters mode | index-filters-loading-1 |
| Search mode, no input | index-filters-loading-2 |
| Search mode, with input | index-filters-loading-3 |

### CODE REVIEW

**Comment by mrcthms** [comment]
@jesstelford I can offer some context around this – we didn't want to have the input flicker lengths when the table is in a  loading state, but rather have a consistent length. I'll double check with @ouellettejordan whether this is something we want to keep, as I know this was an intentional ask when we initially built the component.

File: polaris-react/src/components/Filters/Filters.tsx [hunk 1]
```diff
@@ -283,9 +283,11 @@ export function Filters({
   const additionalContent = useMemo(() => {
     return (
       <>
-        <div className={styles.Spinner}>
-          {loading ? <Spinner size="small" /> : null}
-        </div>
+        {loading ? (
+          <div className={styles.Spinner}>
```
Comment (chloerice) [inline polaris-react/src/components/Filters/Filters.tsx]: What if the spinner replaced the input clear button when loading is true, or rendered to the left of the clear button?

|[replacing](https://github.com/Shopify/polaris/commit/16d2a7732fc330ced9accacb338bb422ff8e39e9)|[left of](https://github.com/Shopify/polaris/commit/7c0c5b649c74985e2db9a1cc975a427a1131c9ab)|
|---|---|
|<video src="https://github.com/Shopify/polaris/assets/18447883/08c3493d-f1c8-41ea-8958-d58374bedfd0" />|<video src="https://github.com/Shopify/polaris/assets/18447883/27c4b524-a721-4a32-b2ae-00fee091adea" />|

**Comment by jesstelford** [comment]
Great input @mrcthms & @chloerice!

I've moved the spinner to be next to the close button, what do you think?

index-filters-loading-demo

|  |  |
|---|---|
| Filters mode | index-filters-loading-1 |
| Search mode, no input | index-filters-loading-2 |
| Search mode, with input | index-filters-loading-3 |

**Comment by jesstelford** [comment]
🤔 ~The [CI failures](https://github.com/Shopify/polaris/actions/runs/6400198323/job/17373463250?pr=10808) don't make sense me - I don't get that locally, and `"2"` should be a valid `gap` type... Any ideas?~ Rebase onto `next` fixed it 👍

File: polaris-react/src/components/Filters/components/SearchField/SearchField.tsx [hunk 2]
```diff
@@ -19,6 +21,8 @@ export interface SearchFieldProps {
   placeholder?: string;
   disabled?: boolean;
   borderlessQueryField?: boolean;
+  /** Show a loading spinner to the right of the input */
```
Comment (chloerice) [inline polaris-react/src/components/Filters/components/SearchField/SearchField.tsx:24]:
```suggestion
  /** Show a loading spinner to the right of the input  value */
```

File: .changeset/sixty-fishes-grin.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': patch
+---
+
+[IndexFilters] Loading spinner moved to be a suffix within the search box.
```
Comment (chloerice) [inline .changeset/sixty-fishes-grin.md:5]:
```suggestion
Moved the `Filters` loading spinner to the suffix of the query filter
```

File: polaris-react/src/components/Filters/components/SearchField/SearchField.scss [hunk 4]
```diff
@@ -54,19 +57,32 @@
   outline-offset: var(--p-space-025);
 }
 
-.ClearButton {
+.Suffix {
```
Comment (chloerice) [inline polaris-react/src/components/Filters/components/SearchField/SearchField.scss:60]: ❤️

File: polaris-react/src/components/Filters/components/SearchField/SearchField.tsx [hunk 5]
```diff
@@ -72,21 +77,32 @@ export function SearchField({
...
+                disabled={disabled}
+              >
+                <Text as="span" visuallyHidden>
+                  {i18n.translate('Polaris.Common.clear')}
+                </Text>
+                <Icon source={CircleCancelMinor} tone="subdued" />
+              </UnstyledButton>
+            ) : null}
+          </InlineStack>
+        </div>
```
Comment (chloerice) [inline polaris-react/src/components/Filters/components/SearchField/SearchField.tsx:104]: Let's move the markup for each of the suffix items into variables so there aren't conditionals in the return.

```tsx
const loadingMarkup = loading
  ? (
    <div className={styles.Spinner}>
      <Spinner size="small" />
    </div>
  ) : null;

const clearButtonMarkup = value !== ''
  ? (
    <UnstyledButton
      onClick={() => handleClear()}
      disabled={disabled}
      className={classNames(
        styles.ClearButton,
        focused && styles['ClearButton-focused'],
      )}
    >
      <Text as="span" visuallyHidden>
        {i18n.translate('Polaris.Common.clear')}
      </Text>
      <Icon source={CircleCancelMinor} tone="subdued" />
    </UnstyledButton>
) : null

const suffixMarkup = loading || value !== '' 
  ? (
    <div className={styles.Suffix}>
      <InlineStack gap="200">
        {loadingMarkup}
        {clearButtonMarkup}
      </InlineStack>
    </div>
  ) : null;
  
  
  ....
  ....
  

{suffixMarkup}
```

**Review by chloerice — APPROVED** [review]
Thanks for tackling this Jess!! 🙌🏽

**Comment by annaShop** [comment]
hi @chloerice , could you please update on when this PR will be merged? 
we have a high priority bug https://github.com/Shopify/temp-project-mover-syfiawoo-20250331132715/issues/334 that should be fixed by this merge. 
thnx!

**Comment by chloerice** [comment]
> hi @chloerice , could you please update on when this PR will be merged? we have a high priority bug [Shopify/temp-project-mover-syfiawoo-20250331132715#334](https://github.com/Shopify/temp-project-mover-syfiawoo-20250331132715/issues/334) that should be fixed by this merge. thnx!

Hey @annaShop 👋🏽 The team was away Bursting last week. This will get merged in the next couple days but won't be updated in Shopify/web until the v12 upgrade ships (which will likely be early next week).

**Pull request merged**